### PR TITLE
Cleanup X11 and XRANDR Macros

### PR DESCRIPTION
**Pull request by nolange**
This fixes build with X11 enabled and XRANDR disabled.

### CODE REVIEW

**Comment by Rumi-Larry** [comment]
https://github.com/dolphin-emu/dolphin/pull/8221 I'm guessing this makes that pr obsolete?

**Comment by nolange** [comment]
> #8221 I'm guessing this makes that pr obsolete?

Yeah, I guess too.

File: Source/Core/DolphinQt/MainWindow.cpp [hunk 1]
```diff
@@ -1463,7 +1463,7 @@ void MainWindow::UpdateScreenSaverInhibition()
 
   m_is_screensaver_inhibited = inhibit;
 
-#if defined(HAVE_XRANDR) && HAVE_XRANDR
+#if defined(HAVE_X11) && HAVE_X11
```
Comment (leoetlino) [inline Source/Core/DolphinQt/MainWindow.cpp:1466]: Could you simplify this to `#ifdef HAVE_X11`? We never define it to 0.

We should probably do the same thing for HAVE_XRANDR after removing this line from the CMakeLists.txt: https://github.com/dolphin-emu/dolphin/blob/ddacbf83f65b9642f3c9cee9e075b7ec3d473321/CMakeLists.txt#L444

File: Source/Core/UICommon/UICommon.cpp [hunk 2]
```diff
@@ -39,7 +39,7 @@
 #include "UICommon/UICommon.h"
 #include "UICommon/USBUtils.h"
 
-#if defined(HAVE_XRANDR) && HAVE_XRANDR
+#if defined(HAVE_X11) && HAVE_X11
```
Comment (leoetlino) [inline Source/Core/UICommon/UICommon.cpp:42]: `#ifdef HAVE_X11`

File: Source/Core/UICommon/UICommon.cpp [hunk 3]
```diff
@@ -394,7 +394,7 @@ bool TriggerSTMPowerEvent()
   return true;
 }
 
-#if defined(HAVE_XRANDR) && HAVE_XRANDR
+#if defined(HAVE_X11) && HAVE_X11
```
Comment (leoetlino) [inline Source/Core/UICommon/UICommon.cpp:397]: `#ifdef HAVE_X11`

File: Source/Core/UICommon/UICommon.h [hunk 4]
```diff
@@ -13,7 +13,7 @@ namespace UICommon
 void Init();
 void Shutdown();
 
-#if defined(HAVE_XRANDR) && HAVE_XRANDR
+#if defined(HAVE_X11) && HAVE_X11
```
Comment (leoetlino) [inline Source/Core/UICommon/UICommon.h:16]: `#ifdef HAVE_X11`

File: Source/Core/DolphinQt/MainWindow.cpp (hunk 1 above)
Comment (nolange) [reply Source/Core/DolphinQt/MainWindow.cpp]: Ok, did add another commit cleaning up the HAVE_XRANDR guards

**Review by leoetlino — APPROVED** [review]

**Pull request merged**